### PR TITLE
Add @implementation to parser

### PR DESCRIPTION
**Pull request by beccadax**
Matches apple/swift#72596.

### CODE REVIEW

**Comment by beccadax** [comment]
With apple/swift#72596

@swift-ci please test

File: CodeGeneration/Sources/SyntaxSupport/KeywordSpec.swift [hunk 1]
```diff
@@ -550,6 +551,8 @@ public enum Keyword: CaseIterable {
       return KeywordSpec("higherThan")
     case .if:
       return KeywordSpec("if", isLexerClassified: true)
+    case .implementation:
+      return KeywordSpec("implementation")
```
Comment (ahoppen) [inline CodeGeneration/Sources/SyntaxSupport/KeywordSpec.swift:555]: Could you mark the keyword as experimental, eg. like `_borrowing`. For that you’ll need to plumb the experimental feature through.
- Define it in https://github.com/apple/swift-syntax/blob/main/CodeGeneration/Sources/SyntaxSupport/ExperimentalFeatures.swift
- Map it in https://github.com/apple/swift/blob/f81f7f9c544abb362f148589bc18609591c0952d/lib/ASTGen/Sources/ASTGen/SourceFile.swift#L53

Comment (beccadax) [reply CodeGeneration/Sources/SyntaxSupport/KeywordSpec.swift:564]: Sure, will do.

**Comment by beccadax** [comment]
In further discussions with @ahoppen, it became clear that I don't really need to modify SwiftSyntax for this—we can parse it as a custom attribute and convert it to our own attribute in ASTGen. Gonna close this PR and do that instead.

**Pull request closed without merging**